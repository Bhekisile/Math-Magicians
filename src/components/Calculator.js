import { useState } from 'react';
import Screen from './Screen';
import calculate from '../logic/calculate';
import Buttons from './Buttons';

export default function Calculator() {
  const [displayValue, setDisplayValue] = useState('0');
  const [calcData, setCalcData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newData = calculate(calcData, buttonName);
    setCalcData(newData);

    if (newData.next !== null) {
      setDisplayValue(newData.next);
    } else if (newData.total !== null) {
      setDisplayValue(newData.total);
    } else {
      setDisplayValue('0');
    }
  };

  return (
    <div className="wrapper">
      <Screen displayValue={displayValue} />
      <Buttons onButtonClick={handleButtonClick} />
    </div>
  );
}
