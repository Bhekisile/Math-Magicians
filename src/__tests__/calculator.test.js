import { fireEvent, render } from '@testing-library/react';
import Calculator from '../components/Calculator';
// import CalculatorParts from '../components/CalculatorParts';

describe('Calculator Component', () => {
  it('renders the calculator with correct heading', () => {
    const { container, getByText } = render(<Calculator />);
    fireEvent.click(getByText('1'));
    expect(container.firstChild).toMatchSnapshot();
  });
});
