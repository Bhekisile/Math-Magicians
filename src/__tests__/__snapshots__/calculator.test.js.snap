// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component renders the calculator with correct heading 1`] = `
<div
  class="content"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    class="calc"
  >
    <div
      class="screen"
    >
      <p>
        1
      </p>
    </div>
    <div
      class="button"
    >
      <button
        class="btn"
        name="AC"
        type="button"
      >
        AC
      </button>
      <button
        class="btn"
        name="+/-"
        type="button"
      >
        +/-
      </button>
      <button
        class="btn"
        name="%"
        type="button"
      >
        %
      </button>
      <button
        class="btn orange"
        name="รท"
        type="button"
      >
        รท
      </button>
      <br />
      <button
        class="btn"
        name="7"
        type="button"
      >
        7
      </button>
      <button
        class="btn"
        name="8"
        type="button"
      >
        8
      </button>
      <button
        class="btn"
        name="9"
        type="button"
      >
        9
      </button>
      <button
        class="btn orange"
        name="x"
        type="button"
      >
        x
      </button>
      <br />
      <button
        class="btn"
        name="4"
        type="button"
      >
        4
      </button>
      <button
        class="btn"
        name="5"
        type="button"
      >
        5
      </button>
      <button
        class="btn"
        name="6"
        type="button"
      >
        6
      </button>
      <button
        class="btn orange"
        name="-"
        type="button"
      >
        -
      </button>
      <br />
      <button
        class="btn"
        name="1"
        type="button"
      >
        1
      </button>
      <button
        class="btn"
        name="2"
        type="button"
      >
        2
      </button>
      <button
        class="btn"
        name="3"
        type="button"
      >
        3
      </button>
      <button
        class="btn orange"
        name="+"
        type="button"
      >
        +
      </button>
      <br />
      <button
        class="zero"
        name="0"
        type="button"
      >
        0
      </button>
      <button
        class="btn"
        name="."
        type="button"
      >
        .
      </button>
      <button
        class="btn orange"
        name="="
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
